src/components/SubscriptionListItem.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+	<ListItem :title="getTitle()"
+		:details="getDetails()">
+		<template #icon>
+			<Avatar :size="44"
+				:url="getImageSrc()"
+				:display-name="getAvatarName()" />
+		</template>
+		<template #subtitle>
+			<span v-if="isLoading"><em>(Loading RSS data...)</em></span>
+			<span v-else>{{ getSubtitle() }}</span>
+		</template>
+		<template #actions>
+			<ActionLink :href="getHomepageLink()"
+				target="_blank"
+				icon="icon-external">
+				Podcast's homepage
+			</ActionLink>
+			<ActionLink :href="getRssLink()"
+				target="_blank">
+				<template #icon>
+					<Rss />
+				</template>
+				RSS feed
+			</ActionLink>
+		</template>
+	</ListItem>
+</template>
+
+<script>
+import ActionLink from '@nextcloud/vue/dist/Components/ActionLink'
+import Avatar from '@nextcloud/vue/dist/Components/Avatar'
+import ListItem from '@nextcloud/vue/dist/Components/ListItem'
+
+import Rss from 'vue-material-design-icons/Rss.vue'
+
+import { generateUrl } from '@nextcloud/router'
+import axios from '@nextcloud/axios'
+
+export default {
+	name: 'SubscriptionListItem',
+	components: {
+		ActionLink,
+		Avatar,
+		ListItem,
+		Rss,
+	},
+	props: {
+		sub: {
+			type: Object,
+			required: true,
+		},
+	},
+	data() {
+		return {
+			podcastData: null,
+			isLoading: true,
+		}
+	},
+	async mounted() {
+		try {
+			const resp = await axios.get(generateUrl('/apps/gpoddersync/personal_settings/podcast_data?url={url}', {
```

Comment:
please extract dependency to axios

Reply:
I've never done dependency injection stuff in Vue. Do you have any recommendations?

Like what folder should the "personalsettings service" be in? And how do I inject it properly?

Reply:
You can just import and use extracted functions

Reply:
Even better would be we pass the fetched data into the component and do not fetch it on mount

Reply:
Pass subscriptions as props and not through a function call onMounted